### ISSUE TITLE
stopping early

### ISSUE THREAD

**Issue by weijiacheng00** [icc2115/Neural-GC#15]
clstm can achieve early stopping on simulated data, but the loss keeps decreasing on real datasets and cannot achieve early stopping.

**Comment by weijiacheng00** [comment]
Perhaps there were no early stops under certain lam conditions? Need to go to max_iter?

**Comment by iancovert** [comment]
I think it may be expected not to encounter early stopping under certain conditions. Our experiments used only a training dataset, no held-out validation set, and it's possible that the model could keep improving through the max number of iterations, including possibly overfitting to the training data. (I would not design experiments that way today, but it's what we did at the time with the datasets we studied.)

For you, it may currently be hard to tell the difference between optimization progress due to legitimate improvements vs overfitting. Perhaps you can tell this by using a held-out set for early stopping detection? If this held-out loss continues to decrease through to the max number of iterations, the longer training duration seems necessary.

Let me know if that makes sense.
